pkg/reconciler/delivery/delivery.go

```diff
@@ -109,133 +114,73 @@ func (c *Reconciler) updateRoute(ctx context.Context, cfg *v1.Configuration) err
 	route := r.DeepCopy()
 	latestReady := cfg.Status.LatestReadyRevisionName
 
-	if isRouteStatusUpToDate(route, latestReady, &policy) {
-		return nil
-	}
-	route, err = modifyRouteSpec(route, latestReady, &policy)
+	route, err = modifyRouteSpec(route, c.revisionLister, latestReady, &policy)
 	if err != nil {
 		return err
 	}
 
-	logger.Info("Applying updated Route object")
-
-	// record the timestamp for the current udpate to the Route object before actually pushing it
-	// this is used later when determining if Route status is up to date
-	if route.Annotations == nil {
-		route.Annotations = make(map[string]string)
-	}
-	route.Annotations[AnnotationKey] = c.timeProvider().Format(TimeFormat)
+	logger.Info("Applying Route object")
 	_, err = c.client.ServingV1().Routes(cfg.Namespace).Update(route)
 	if err != nil {
 		return err
 	}
 
 	// when we have latestRevision = true, we know that we don't need to queue future events
 	if *route.Spec.Traffic[0].LatestRevision {
-		logger.Info("Progressive rollout completed!")
+		logger.Info("Routing state has stabilized!")
 		return nil
 	}
-	t, e := getThreshold(&policy, int(*route.Spec.Traffic[1].Percent))
-	if e != nil {
-		return e
-	}
-	logger.Infof("Queueing event for %v seconds later", t)
-	c.followup(cfg, time.Duration(t) * time.Second)
 
-	return nil
-}
-
-// isRouteStatusUpToDate determines if the current Route status already matches our desired state
-func isRouteStatusUpToDate(route *v1.Route, newRevName string, policy *Policy) bool {
-	// the Route status is up to date if:
-	// 1. the new Revision is listed in the status traffic targets, AND
-	// 2. the Route time stamp hasn't expired
-	// OR if:
-	// 3. the new Revision is listed in the status traffic targets, AND
-	// 4. the new Revision already reached 100%
-	nameListed := false
-	for _, t := range route.Status.Traffic {
-		if t.RevisionName == newRevName {
-			nameListed = true
-			break
-		}
-	}
-	if !nameListed {
-		return false
-	}
-	if len(route.Status.Traffic) == 1 || *route.Status.Traffic[1].Percent == 100 {
-		return true
+	// print (Revision, Percent) pairs in a VERY friendly format into the logs for convenient viewing
+	fmt.Println()
+	for _, t := range route.Spec.Traffic {
+		fmt.Printf("%v: %v\n", t.RevisionName, *t.Percent)
 	}
-	// by design, accessing route.Annotations[AnnotationKey] should not cause error
-	previousTime, err := time.Parse(TimeFormat, route.Annotations[AnnotationKey])
+	fmt.Println()
```

Comment:
use logger?
Also do not checkin debug logging.

Reply:
Oh sorry, I put this in for my own debugging purposes. I should have deleted it but I forgot!

Reply:
Still here partially

Reply:
deleted